lib/brakeman/checks/check_sql.rb

```diff
@@ -579,6 +579,10 @@ def check_string_arg exp
     :where_values_hash, :foreign_key, :uuid
   ]
 
+  def ignore_methods_in_sql
+    @ignore_methods_in_sql ||= IGNORE_METHODS_IN_SQL.merge(tracker.options[:sql_safe_methods] || [])
```

Comment:
@escheiner - here is the problem!

`IGNORE_METHODS_IN_SQL` is a `Set`, and `Set#merge` is a destructive operation:

```ruby
2.7.0 :001 > s = Set.new
2.7.0 :002 > s.merge [:a]
 => #<Set: {:a}> 
2.7.0 :003 > s
 => #<Set: {:a}>
```

I think `Set#+` will provide the expected behavior:

```ruby
2.7.0 :001 > s = Set[:a, :b]
2.7.0 :002 > s
 => #<Set: {:a, :b}> 
2.7.0 :003 > s + [:c]
 => #<Set: {:a, :b, :c}> 
2.7.0 :004 > s
 => #<Set: {:a, :b}>
```